packages/palette/src/elements/LineChart/LineChart.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useRef, useState } from "react"
+import styled from "styled-components"
+import { space } from "../../helpers"
+import { useHasEnteredViewport } from "../../helpers/visualizationHelpers"
+import { coerceLabel } from "../BarChart"
+import { BarHoverLabel } from "../BarChart/Bar"
+import { BarLabelProps } from "../BarChart/BarLabel"
+import { ProvideMousePosition } from "../BarChart/MousePositionContext"
+import { Flex } from "../Flex"
+import { Sans } from "../Typography"
+import { LineChartSVG } from "./LineChartSVG"
+
+export interface PointDescriptor {
+  value: number
+  axisLabelX?: React.ReactNode
+  tooltip?: React.ReactNode | BarLabelProps
+}
+
+export interface LineChartProps {
+  points: PointDescriptor[]
+  chartHeight?: number
+}
+
+const margin = space(2)
+const DEFAULT_HEIGHT = 87
+
+/**
+ * LineChart is a component that displays some data points connected by lines.
+ * Useful for visualizing a time series, etc.
+ */
+export const LineChart = ({ points, chartHeight }: LineChartProps) => {
+  const [width, setWidth] = useState(0)
+  const [height, setHeight] = useState(DEFAULT_HEIGHT)
+  const [hoverIndex, setHoverIndex] = useState(-1)
+
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const setContainerWidth = () => {
+    if (wrapperRef.current) {
+      setWidth(wrapperRef.current.getBoundingClientRect().width - 5)
+    }
+  }
+
+  if (!width) {
+    setContainerWidth()
+  }
+
+  useEffect(() => {
+    setHeight(chartHeight || DEFAULT_HEIGHT)
+  }, [chartHeight])
+
+  useEffect(() => {
+    window.addEventListener("resize", setContainerWidth)
+
+    return function cleanup() {
+      window.removeEventListener("resize", setContainerWidth)
+    }
+  }, [])
+
+  const hasEnteredViewport = useHasEnteredViewport(wrapperRef)
```

Comment:
The width / height code that's above this line can be shared with the other chart, yah?

Reply:
Maybe for the donut chart if we end up adding it to palette. `BarChart` doesn't need that because it doesn't have `SVG`.